builder/builder.go

```diff
@@ -692,19 +1057,18 @@ func (b *Builder) BuildUpdate(prefixflag string, minVersion int, format string,
 	timer.Start("MINIMIZE STORED CHROOTS")
 	// Clean up the bundle chroots as only the full chroot is needed from this point on.
 	if !keepChroots {
-		var files []os.FileInfo
-		files, err = ioutil.ReadDir(b.BundleDir)
+		// Get the set of bundles for which chroots were built
+		var set bundleSet
+		set, err = b.getFullMixBundleSet()
```

Comment:
Optional: should we just move this one up (before chroot building) so that it gets used there (for the new codepath) and here (for both new and old)?

Reply:
... and validation might be worth running for both new and old codepaths (for chroot building) so extract that as well.

Reply:
I'm not following this yet. You're commenting on the line in `BuildUpdate`, where it gets the full list of mix bundles so that it can clean up old chroots. What are you proposing could be moved up? Maybe you're misunderstanding what `getFullMixBundleSet` does? It's only building up the list in memory...

Reply:
Never mind. I'm just mixing up the two commands :-)